src/LazyClient.php

```diff
@@ -52,66 +59,100 @@ private function client()
+                        $idleTimer = null;
+                    }
+                });
 
-            Util::forwardEvents(
-                $client,
-                $self,
-                array(
-                    'message',
-                    'subscribe',
-                    'unsubscribe',
-                    'pmessage',
-                    'psubscribe',
-                    'punsubscribe',
-                )
-            );
+                // keep track of all channels and patterns this connection is subscribed to
+                $client->on('subscribe', function ($channel) use (&$subscribed) {
+                    $subscribed[$channel] = true;
+                });
+                $client->on('psubscribe', function ($pattern) use (&$psubscribed) {
+                    $psubscribed[$pattern] = true;
+                });
+                $client->on('unsubscribe', function ($channel) use (&$subscribed) {
+                    unset($subscribed[$channel]);
+                });
+                $client->on('punsubscribe', function ($pattern) use (&$psubscribed) {
+                    unset($psubscribed[$pattern]);
+                });
+
+                Util::forwardEvents(
+                    $client,
+                    $self,
+                    array(
+                        'message',
+                        'subscribe',
+                        'unsubscribe',
+                        'pmessage',
+                        'psubscribe',
+                        'punsubscribe',
+                    )
+                );
 
-            return $client;
-        }, function (\Exception $e) use (&$pending) {
-            // connection failed => discard connection attempt
-            $pending = null;
+                return $client;
+            }, function (\Exception $e) use (&$pending) {
+                // connection failed => discard connection attempt
+                $pending = null;
 
-            throw $e;
+                throw $e;
+            });
         });
     }
 
+    private function handleSentinelConnection()
+    {
+        $master_name =& $this->master_name;
+
+        $defer = new Deferred();
+        if (null !== $master_name) {
+            $connector = new TcpConnector($this->loop);
+
+            $connector->connect($this->target)
+                ->then(function (ConnectionInterface $connection) use ($defer, $master_name) {
+                    $connection->on('data', function($chunk) use ($defer) {
+                        $parser = new ResponseParser();
+                        /** @var MultiBulkReply $reply */
+                        $reply = $parser->pushIncoming($chunk);
+                        /** @var array<int, int|float|string|bool> $data */
+                        $data = current($reply)->getValueNative();
+
+                        $defer->resolve("$data[0]:$data[1]");
+                    });
+
+                    $connection->write("SENTINEL get-master-addr-by-name {$master_name} \r\n");
```

Comment:
This should probably use the client API and call the sentinel command like this

https://github.com/clue/reactphp-redis/blob/426e9dfd71e7ffa20de080a725589ff9b42762ca/src/Factory.php#L99